### PR TITLE
prefix [linter_name] to all human-readable output formats

### LINKED ISSUE #1357: Feature request: include linter name in warnings
## Request

Consider the following code:

```r
library(lintr)
f <- tempfile(fileext = ".R")
writeLines("do_thing <- T", f)
lintr::lint(
    filename = f
    , linters = list(
        "tf" = lintr::T_and_F_symbol_linter
    )
    , cache = FALSE
)
```

On `{lintr}` 2.0.1 installed from CRAN and on the latest version from GitHub (https://github.com/r-lib/lintr/commit/801fac322613121f32ae916af379161a838e3b27, installed with `remotes::install_github("r-lib/lintr")`), this produces the following output (the `(...)` is mine, just to make the path a bit more readable):

```text
(...)/T/RtmpIvPGWZ/fileb817b0f6442.R:1:14: style: Use TRUE instead of the symbol T.
do_thing <- T
            ~^
```

As a user of `{lintr}`, I'd like to see the linter name printed in the output, like this:

```text
(...)/RtmpIvPGWZ/fileb817b0f6442.R:1:14: style: [T_and_F_symbol_linter] Use TRUE instead of the symbol T.
do_thing <- T
            ~^
```

## Motivation

I learned tonight from @MichaelChirico that as of `{lintr}` 3.0.0, it'll be possible to use a comment like the following to say "ignore warnings from a specific linter on this line" (https://github.com/microsoft/LightGBM/pull/5249#discussion_r889447818).

```r
x <- T  # nolint: T_and_F_symbol_linter`
```

I totally see the value in being specific there and would like to do that. But since the linter name isn't printed in the warnings from `lintr::lint()`, figuring out the right value to add in that comment will require some trial-and-error or searching in this package's source code.

## References:

As an example, `flake8` also supports such targeted comments (https://flake8.pycqa.org/en/3.1.1/user/ignoring-errors.html#in-line-ignoring-errors), and I've found it really helpful that it prints in its output exactly the code that should be added to `# noqa` comments.

```shell
echo "x = (1,2,3,y)" > ./out.py
flake8 ./out.py
```

```text
./out.py:1:7: E231 missing whitespace after ','
./out.py:1:9: E231 missing whitespace after ','
./out.py:1:11: E231 missing whitespace after ','
./out.py:1:12: F821 undefined name 'y'
```

Example suppressing one specific warning:

```shell
echo "x = (1,2,3,y)  # noqa: E231" > ./out.py
flake8 ./out.py
```

```text
./out.py:1:12: F821 undefined name 'y'
```

Thanks for your time and consideration!

### PR DESCRIPTION
**Pull request by AshesITR**
fixes #1357 

Didn't touch `checkstyle_output()`.

CC @renkun-ken Not sure if this is relevant to the vscode linter support.

### CODE REVIEW

File: R/actions.R [hunk 1]
```diff
@@ -13,7 +13,7 @@ github_actions_log_lints <- function(lints, project_dir = "") {
     )
     cat(sprintf(
       "::warning %s::%s,%s\n",
```
Comment (MichaelChirico) [inline R/actions.R:15]: why not adjust the template instead?

File: tests/testthat/test-rstudio_markers.R [hunk 2]
```diff
@@ -4,47 +4,56 @@ test_that("it returns markers which match lints", {
...
+  lint1[[1L]]$linter <- "linter_name"
 
   marker1 <- rstudio_source_markers(lint1)
   expect_equal(marker1$name, "lintr")
   expect_equal(marker1$markers[[1L]]$type, lint1[[1L]]$type)
   expect_equal(marker1$markers[[1L]]$file, lint1[[1L]]$filename)
   expect_equal(marker1$markers[[1L]]$line, lint1[[1L]]$line_number)
   expect_equal(marker1$markers[[1L]]$column, lint1[[1L]]$column_number)
-  expect_equal(marker1$markers[[1L]]$message, lint1[[1L]]$message)
+  expect_equal(marker1$markers[[1L]]$message, paste0("[", lint1[[1L]]$linter, "] ", lint1[[1L]]$message))
```
Comment (MichaelChirico) [inline tests/testthat/test-rstudio_markers.R:26]: we really need some better test helpers... really hard to tell what the impact of the change is exactly.

something like `expect_lint_output(x, "...", rstudio = TRUE)`, where we do some `trim_some()`-like logic on `"..."` to maximize readability would make it super clear to the test reader and super easy for the test writer.

(for another day...)

**Comment by MichaelChirico** [comment]
would you mind attaching a screenshot of the new output if you get a chance? i haven't had a chance to pull & install to check myself

**Comment by AshesITR** [comment]
I don't use RStudio, but here is the console output:

image

**Comment by MichaelChirico** [comment]
Code changes LGTM, but I've a terrible eye for design-y stuff. Let's get a third pair of eyes on it... @renkun-ken or @jameslamb care to weigh in?

**Comment by jameslamb** [comment]
> care to weigh in?

I'm not familiar enough with the internals of `{lintr}` to comment on specific code changes, but happy to say that https://github.com/r-lib/lintr/pull/1363#issuecomment-1146888918 looks like exactly what I was envisioning. As long as I can rely on putting whatever is in `[]` into a `# nolint: ` comment to suppress that specific warning, looks great to me!

**Comment by MichaelChirico** [comment]
no need to know internals, only asking about the end-user appearance part.

and good enough for me!

**Review by MichaelChirico — APPROVED** [review]

**Pull request merged**